### PR TITLE
Hide temp place name button, hide places table, hide places wizard step.

### PR DESCRIPTION
**Pull request by kasenvr**

### CODE REVIEW

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-966eea2.tar_2e75166e.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-966eea2_2e75166e.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-966eea2.tar_2e75166e.gz

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-966eea2_2e75166e.exe

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-966eea2_2e75166e.dmg

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-3be0d2b.tar_25466391.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-3be0d2b.tar_25466391.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-3be0d2b_25466391.sh

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-3be0d2b_25466391.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-3be0d2b_25466391.exe

File: domain-server/resources/web/js/shared.js [hunk 1]
```diff
@@ -500,6 +500,33 @@ function prepareAccessTokenPrompt(callback) {
   });
 }
 
+function createDomainIDPrompt(callback) {
+    swal({
```
Comment (ctrlaltdavid) [inline domain-server/resources/web/js/shared.js:504]:
```suggestion
  swal(
    {
```

File: domain-server/resources/web/js/shared.js [hunk 2]
```diff
@@ -500,6 +500,33 @@ function prepareAccessTokenPrompt(callback) {
...
+      title: 'Finish Registering Domain',
+      type: 'input',
+      text: 'Enter a label for this machine.</br></br>This will help you identify which domain ID belongs to which machine.</br></br>This is a required step for registration.</br></br>',
+      showCancelButton: true,
+      confirmButtonText: "Create",
+      closeOnConfirm: false,
+      html: true
+    }, function(inputValue){
```
Comment (ctrlaltdavid) [inline domain-server/resources/web/js/shared.js:512]:
```suggestion
    }, function(inputValue) {
```

File: domain-server/resources/web/js/shared.js [hunk 3]
```diff
@@ -500,6 +500,33 @@ function prepareAccessTokenPrompt(callback) {
...
+      if (inputValue === false) {
+      return false;
+    }
```
Comment (ctrlaltdavid) [inline domain-server/resources/web/js/shared.js:515]: Indent here and in general in this function is a bit off and confusing.

File: domain-server/resources/web/js/shared.js [hunk 4]
```diff
@@ -500,6 +500,33 @@ function prepareAccessTokenPrompt(callback) {
...
+
+    if (inputValue === "") {
+      swal.showInputError("Please enter a valid label for your machine.")
+      return false
```
Comment (ctrlaltdavid) [inline domain-server/resources/web/js/shared.js:519]:
```suggestion
      swal.showInputError("Please enter a valid label for your machine.");
      return false;
```

File: domain-server/resources/web/wizard/js/wizard.js [hunk 5]
```diff
@@ -171,6 +172,34 @@ function setupWizardSteps() {
   $(currentStep).show();
 }
 
+function promptToCreateDomainID() {
+    setTimeout(function(){ 
+        createDomainIDPrompt(function(label) {
+            var domainJSON = {
+              "label": label
+            }
```
Comment (ctrlaltdavid) [inline domain-server/resources/web/wizard/js/wizard.js:180]: Inconsistent indenting.

File: domain-server/resources/web/wizard/js/wizard.js [hunk 6]
```diff
@@ -171,6 +172,34 @@ function setupWizardSteps() {
...
+        
+            $.post("/api/domains", domainJSON, function(data){
```
Comment (ctrlaltdavid) [inline domain-server/resources/web/wizard/js/wizard.js:182]:
```suggestion
            $.post("/api/domains", domainJSON, function(data) {
```

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

File: domain-server/resources/web/wizard/js/wizard.js (hunk 5 above)
Comment (kasenvr) [reply domain-server/resources/web/wizard/js/wizard.js:180]: I'm going to use Atom to indent all these files to 4 lines after this, I feel like that'll be a better solution so no one has to deal with that again? :)

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-d149f70.tar_5725a322.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-d149f70_5725a322.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-d149f70.tar_5725a322.Z

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-d149f70_5725a322.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-d149f70_5725a322.exe

File: domain-server/resources/web/wizard/js/wizard.js (hunk 5 above)
Comment (ctrlaltdavid) [reply domain-server/resources/web/wizard/js/wizard.js:180]: There are pros and cons to reworking all the indenting. A con is that you lose git blame history for each line.

Comment (kasenvr) [reply domain-server/resources/web/wizard/js/wizard.js:180]: Ah, darn it. Then I'm not sure anymore, it drives me mildly crazy though every time I go in there to fix something. Hmph. I'll fix it here then.

File: domain-server/resources/web/js/shared.js [hunk 7]
```diff
@@ -501,17 +501,16 @@ function prepareAccessTokenPrompt(callback) {
...
-      if (inputValue === false) {
+  swal({
+    title: 'Finish Registering Domain',
+    type: 'input',
+    text: 'Enter a label for this machine.</br></br>This will help you identify which domain ID belongs to which machine.</br></br>This is a required step for registration.</br></br>',
+    showCancelButton: true,
+    confirmButtonText: "Create",
+    closeOnConfirm: false,
+    html: true
+  }, function(inputValue) {
```
Comment (ctrlaltdavid) [inline domain-server/resources/web/js/shared.js:512]:
```suggestion
  }, function (inputValue) {
```

File: domain-server/resources/web/wizard/js/wizard.js [hunk 8]
```diff
@@ -173,31 +173,31 @@ function setupWizardSteps() {
...
-            
-              // POST the form JSON to the domain-server settings.json endpoint so the settings are saved
-              postSettings(formJSON, goToNextStep);
-            }, 'json').fail(function(){
-              console.log("Failed to create domain ID...");
-              goToNextStep();
-            });
-        });
-    }, 500); // Apparently swal needs time before opening another prompt.
+  setTimeout(function() {
```
Comment (ctrlaltdavid) [inline domain-server/resources/web/wizard/js/wizard.js:176]:
```suggestion
  setTimeout(function () {
```

File: domain-server/resources/web/wizard/js/wizard.js [hunk 9]
```diff
@@ -173,31 +173,31 @@ function setupWizardSteps() {
...
+    createDomainIDPrompt(function(label) {
```
Comment (ctrlaltdavid) [inline domain-server/resources/web/wizard/js/wizard.js:177]:
```suggestion
    createDomainIDPrompt(function (label) {
```

File: domain-server/resources/web/wizard/js/wizard.js [hunk 10]
```diff
@@ -173,31 +173,31 @@ function setupWizardSteps() {
...
+      var domainJSON = {
+        "label": label
+      }
```
Comment (ctrlaltdavid) [inline domain-server/resources/web/wizard/js/wizard.js:180]:
```suggestion
      };
```

File: domain-server/resources/web/wizard/js/wizard.js [hunk 11]
```diff
@@ -173,31 +173,31 @@ function setupWizardSteps() {
...
+
+      $.post("/api/domains", domainJSON, function(data) {
```
Comment (ctrlaltdavid) [inline domain-server/resources/web/wizard/js/wizard.js:182]:
```suggestion
      $.post("/api/domains", domainJSON, function (data) {
```

File: domain-server/resources/web/wizard/js/wizard.js [hunk 12]
```diff
@@ -173,31 +173,31 @@ function setupWizardSteps() {
...
+        // we successfully created a domain ID, set it on that field
+        var domainID = data.domain.domainId;
+        console.log("Setting domain ID to ", data, domainID);
+
+        var formJSON = {
+          "metaverse": {
+            "id": domainID
+          }
+        }
```
Comment (ctrlaltdavid) [inline domain-server/resources/web/wizard/js/wizard.js:191]:
```suggestion
        };
```

File: domain-server/resources/web/wizard/js/wizard.js [hunk 13]
```diff
@@ -173,31 +173,31 @@ function setupWizardSteps() {
...
+
+        // POST the form JSON to the domain-server settings.json endpoint so the settings are saved
+        postSettings(formJSON, goToNextStep);
+      }, 'json').fail(function() {
```
Comment (ctrlaltdavid) [inline domain-server/resources/web/wizard/js/wizard.js:195]:
```suggestion
      }, 'json').fail(function () {
```

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-6506749_69fe4713.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-6506749_69fe4713.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-6506749.tar_69fe4713.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-6506749.tar_69fe4713.gz

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-6506749_69fe4713.exe

**Review by ctrlaltdavid — APPROVED** [review]

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-e9ba989_69fe4713.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-e9ba989.tar_69fe4713.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-e9ba989.tar_69fe4713.Z

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-e9ba989_69fe4713.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-e9ba989_69fe4713.exe

**Comment by Misterblue** [comment]
This PR is working for me. If the domain starts without --get-temp-name, the domain starts up without talking to the metaverse-server. The connect account then starts the domain talking. Setting the domain name works (see note below) and removing the place editing fixed that problem.

The only problem I noticed was if the connection dialogs get errors (like badly formed domain name), the dialog just exited with no indication to the user.

Two recommondations:
1) Note in 'name domain' dialog that says a domain name can only be alpha-numeric (upper-lower case and "+-_".
2) Error recovery in the login and domain naming dialogs.

There are other things on the settings page that needs work, but I'll make an issue on the domain-server.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-2b38770.tar_c1a47878.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-2b38770.tar_c1a47878.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-2b38770_c1a47878.sh

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-2b38770_c1a47878.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-2b38770_c1a47878.exe

**Review by ctrlaltdavid — APPROVED** [review]

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-2308fb5.tar_70b04b12.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-2308fb5_70b04b12.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-2308fb5.tar_70b04b12.gz

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-2308fb5_70b04b12.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR803/Vircadia-Alpha-PR803-2308fb5_70b04b12.exe

File: domain-server/resources/web/settings/js/settings.js [hunk 14]
```diff
@@ -711,8 +713,8 @@ $(document).ready(function(){
       name: 'places',
       label: 'Places',
       html_id: Settings.PLACES_TABLE_ID,
-      help: "The following places currently point to this domain.</br>To point places to this domain, "
-        + " go to the <a href='" + METAVERSE_URL + "/user/places'>My Places</a> "
+      help: "To point places to this domain, "
+        + " go to the <a href='" + METAVERSE_URL + "/user/places'>Places</a> "
```
Comment (ctrlaltdavid) [inline domain-server/resources/web/settings/js/settings.js:717]: Should this URL be https://dashboard.vircadia.com/places?

**Comment by ctrlaltdavid** [comment]
Tested the PR build...

Step 1 of 3: Clicked "Connect your Metaverse account" button.
	Opened Web page.
	Clicked "Generate" button.
	Copied token.
	"Finish Registering Domain" dialog: CAD-PR803-Domain-Label
Step 2 of 3: Add your Metaverse username
	Entered username.
Step 3 of 3: Enable automatic threading.
	Didn't tick.
Finish

No places listed in domain server settings page. Instead, text saying "To point places to this domain, go to the Places page in your Metaverse account."

Steps 1 through Finish unexpectedly creates two places:
- "amber-hyena"
- The domain label entered in step 1, i.e., "CAD-PR803-Domain-Label"

Steps 1 through Finish also create two domain labels: the same as the places.

**Comment by kasenvr** [comment]
I would say that because you didn't hit any actual errors... account connected, domain connected, and you were presumably able to do anything you wanted (including create a place?) then it's a success.

Plenty to fix on the flow, but in general this appears to solve the problem of things just not working in general that many people have been hitting.

File: domain-server/resources/web/settings/js/settings.js (hunk 14 above)
Comment (kasenvr) [reply domain-server/resources/web/settings/js/settings.js:717]: The metaverse API will redirect you when you hit that link, so the metaverse API can specify where its dashboard is.

**Pull request merged**